packetbeat/sniffer/afpacket_linux.go

```diff
@@ -40,6 +43,85 @@ type afpacketHandle struct {
 	promiscPreviousStateDetected bool
 	device                       string
 	log                          *logp.Logger
+	metrics                      *metrics
+}
+
+type metrics struct {
+	unregister func()
+	done       chan struct{} // used to signal to polling goroutine to stop
+
+	device             *monitoring.String // name of the device being monitored
+	socketPackets      *monitoring.Uint   // number of packets delivered by kernel
+	socketDrops        *monitoring.Uint   // number of packets dropped by kernel (i.e., buffer full)
+	socketQueueFreezes *monitoring.Uint   // number of queue freezes
+	packets            *monitoring.Uint   // number of packets read off buffer by packetbeat
+	polls              *monitoring.Uint   // number of blocking syscalls made by packetbeat waiting for packets
+}
+
+func (m *metrics) close() {
+	if m == nil {
+		return
+	}
+	m.unregister()
+	if m.done != nil {
+		close(m.done)
+		m.done = nil
+	}
+}
+
+func newMetrics(id, device string, interval time.Duration, handle *afpacket.TPacket, log *logp.Logger) *metrics {
+	devID := fmt.Sprintf("%s-af_packet::%s", id, device)
```

Comment:
Are we sure this is safe? From memory, all the child sniffers share the same ID which means that we would have a collision and panic if there is more than one child in the config on the same device. People maybe shouldn't be doing that, but we don't prevent it and it would be good to avoid the panic.

Reply:
What sort of situation are we talking about here? There should be a metrics instance associated with every af_packet handle. I could potentially see this being an issue with multiple protocols in use (I'll test that scenario). Would you have an example config of the scenario you are thinking of (more than one child in the config on the same device)?

Do you think something like what you did for `protos/tcp` work? We want to uniquely identify each sniffer instance, and ideally use something we can tie back to configuration. Sure, we could hash a bunch of details together, but then a user would have no idea what it means. So, ideally, we want to use something human-readable.

Reply:
I tried defining multiple interfaces with the same device, and that's not allowed:

```
{"log.level":"error","@timestamp":"2023-06-05T16:17:21.909Z","log.origin":{"file.name":"instance/beat.go","file.line":1274},"message":"Exiting: duplicated device configurations: enp0s3","service.name":"packetbeat","ecs.version":"1.6.0"}
```

I also tried with multiple protocols and no issues there.

Reply:
Cool. Thanks for checking that.